subsys/fs/littlefs_fs.c

```diff
@@ -751,10 +751,77 @@ static const struct fs_file_system_t littlefs_fs = {
 	.statvfs = littlefs_statvfs,
 };
 
+#define DT_DRV_COMPAT zephyr_fstab_littlefs
+#define FS_PROP(inst, prop) DT_PROP(DT_DRV_INST(inst), prop)
+#define FS_MP_NAME(nodeid) _CONCAT(z_fsmp_, nodeid)
+#define FS_PARTITION(inst) DT_PHANDLE_BY_IDX(DT_DRV_INST(inst), partition, 0)
+
+#define DEFINE_FS(inst) \
+static uint8_t __aligned(4) \
+	read_buffer_##inst[DT_INST_PROP(inst, cache_size)]; \
+static uint8_t __aligned(4) \
+	prog_buffer_##inst[DT_INST_PROP(inst, cache_size)]; \
+static uint32_t lookahead_buffer_##inst[DT_INST_PROP(inst, lookahead_size) \
```

Comment:
`BUILD_ASSERT((DT_INST_PROP(inst, lookahead_size) % 8) == 0, "lookahead-size must be a multiple of 8");` or so might be nice.

Reply:
I've replicated all the build asserts.